### PR TITLE
Froxel visualization debug option

### PR DESCRIPTION
**Pull request by pixelflinger**
The setting can be changed at runtime using a debug property.

Screenshot 2023-10-19 at 12 27 02

### CODE REVIEW

File: shaders/src/light_punctual.fs [hunk 1]
```diff
@@ -242,4 +242,29 @@ void evaluatePunctualLights(const MaterialInputs material,
         color.rgb += surfaceShading(pixel, light, visibility);
 #endif
     }
+
+    if (CONFIG_DEBUG_FROXEL_VISUALLIZATION) {
```
Comment (poweifeng) [inline shaders/src/light_punctual.fs:246]:
```suggestion
    if (CONFIG_DEBUG_FROXEL_VISUALIZATION) {
```

File: filament/src/details/Material.cpp [hunk 2]
```diff
@@ -249,6 +249,9 @@ FMaterial::FMaterial(FEngine& engine, const Material::Builder& builder)
     mSpecializationConstants.push_back({
                     +ReservedSpecializationConstants::CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP,
                     (bool)engine.debug.shadowmap.debug_directional_shadowmap });
+    mSpecializationConstants.push_back({
+                    +ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALLIZATION,
```
Comment (poweifeng) [inline filament/src/details/Material.cpp:253]:
```suggestion
                    +ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALIZATION,
```

File: filament/src/details/Material.cpp [hunk 3]
```diff
@@ -453,12 +456,12 @@ FMaterial::~FMaterial() noexcept {
...
-    auto pos = std::find_if(mSpecializationConstants.begin(), mSpecializationConstants.end(),
-            [&](const auto& item) {
-                return item.id == ReservedSpecializationConstants::CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP;
-            });
-    if (pos != mSpecializationConstants.end()) {
-        pos->value = mEngine.debug.shadowmap.debug_directional_shadowmap;
+    for (auto& item : mSpecializationConstants) {
+        if (item.id == ReservedSpecializationConstants::CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP) {
+            item.value = mEngine.debug.shadowmap.debug_directional_shadowmap;
+        } else if (item.id == ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALLIZATION) {
```
Comment (poweifeng) [inline filament/src/details/Material.cpp:462]:
```suggestion
        } else if (item.id == ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALIZATION) {
```

File: libs/filabridge/include/private/filament/EngineEnums.h [hunk 4]
```diff
@@ -65,6 +65,7 @@ enum class ReservedSpecializationConstants : uint8_t {
     CONFIG_FROXEL_BUFFER_HEIGHT = 4,
     CONFIG_POWER_VR_SHADER_WORKAROUNDS = 5,
     CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP = 6,
+    CONFIG_DEBUG_FROXEL_VISUALLIZATION = 7,
```
Comment (poweifeng) [inline libs/filabridge/include/private/filament/EngineEnums.h:68]:
```suggestion
    CONFIG_DEBUG_FROXEL_VISUALIZATION = 7,
```

File: libs/filamat/src/shaders/CodeGenerator.cpp [hunk 5]
```diff
@@ -238,6 +238,10 @@ utils::io::sstream& CodeGenerator::generateProlog(utils::io::sstream& out, Shade
     generateSpecializationConstant(out, "CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP",
             +ReservedSpecializationConstants::CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP, false);
 
+    // froxel visualization
+    generateSpecializationConstant(out, "CONFIG_DEBUG_FROXEL_VISUALLIZATION",
+            +ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALLIZATION, false);
```
Comment (poweifeng) [inline libs/filamat/src/shaders/CodeGenerator.cpp:243]:
```suggestion
    generateSpecializationConstant(out, "CONFIG_DEBUG_FROXEL_VISUALIZATION",
            +ReservedSpecializationConstants::CONFIG_DEBUG_FROXEL_VISUALIZATION, false);
```

**Review by poweifeng — APPROVED** [review]

File: shaders/src/light_punctual.fs (hunk 1 above)
Comment (pixelflinger) [reply shaders/src/light_punctual.fs]: argh. thanks!

**Pull request merged**